src/main/js/main_html5.js

```diff
@@ -507,18 +508,18 @@ require("../../../html5-common/js/utils/environment.js");
           var captions = closedCaptions[captionsFormat][language];
           var label = captions.name;
           var src = captions.url;
-          var mode = (!!params && params.mode) || 'showing';
 
           $(_video).append("<track class='" + TRACK_CLASS + "' kind='subtitles' label='" + label + "' src='" + src + "' srclang='" + language + "' default>");
           if (_video.textTracks && _video.textTracks[0]) {
-            _video.textTracks[0].mode = mode;
-            if (mode != "showing") {
+            _video.textTracks[0].mode = captionMode;
+            if (captionMode == OO.CONSTANTS.CLOSED_CAPTIONS.HIDDEN) {
               _video.textTracks[0].oncuechange = onClosedCaptionCueChange;
             }
           }
 
           _.delay(function() {
-            if (OO.isFirefox) {
+            if (OO.isFirefox && _video.textTracks && _video.textTracks[0]) {
+              _video.textTracks[0].mode = captionMode;
```

Comment:
possibly silly question - is the caption mode correct at this point?  Should it be passed into this anon function so that it's statically assigned when the function is run?